blockmanager.go

```diff
@@ -425,16 +429,20 @@ func (b *blockManager) handleDonePeerMsg(peers *list.List, p *peer) {
 
 	// Remove requested transactions from the global map so that they will
 	// be fetched from elsewhere next time we get an inv.
-	for k := range p.requestedTxns {
-		delete(b.requestedTxns, k)
+	for k, txPeer := range b.requestedTxns {
+		if txPeer == p { // peer matches
```

Comment:
I believe we have unique IDs for peers already.  We should probably use that instead of relying on pointer equality.  I realize that as long as the code is super careful they should match, but it wouldn't be very difficult to have two different pointers to the same peer which would cause a check like this fail.